python/lsst/summit/utils/guiders/transformation.py

```diff
@@ -0,0 +1,714 @@
+
+    Parameters
+    ----------
+    stamp : array
+        raw ROI array in amp coordinates
+    detector : lsst.afw.cameraGeom.Detector
+        Detector of interest.
+    ampName : str
+        amplifier name, eg. C00
+
+    Returns
+    -------
+    img : array
+        ROI image flipped to be in CCD coordinate orientation
+    """
+    amp = detector[ampName]
+    img = stamp.copy()
+    if amp.getRawFlipX():
+        img = np.fliplr(img)
+    if amp.getRawFlipY():
+        img = np.flipud(img)
+    return img
+
+
+def convert_to_focal_plane(xccd: np.ndarray, yccd: np.ndarray, detNum: int) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Convert from ccd pixels coordinates to focal plane coordinates.
+
+    Parameters
+    ----------
+    xccd : np.ndarray
+        Array of x CCD pixel coordinates.
+    yccd : np.ndarray
+        Array of y CCD pixel coordinates.
+    detNum : int
+        Detector number (e.g., 189 for R22_S11).
+
+    Returns
+    -------
+    xfp, yfp : np.ndarray
+        Arrays of x and y focal plane coordinates (mm).
+
+    """
+    if len(xccd) > 0:
+        detector = LsstCam.getCamera()[detNum]
+        # Convert the star positions to focal plane coordinates
+        xfp, yfp = pixel_to_focal(xccd, yccd, detector)
+    else:
+        xfp, yfp = np.array([]), np.array([])
+    return xfp, yfp
+
+
+def convert_to_altaz(
+    xccd: np.ndarray, yccd: np.ndarray, wcs: Any, obs_time: Time
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Convert the ccd pixel coordinates to altaz coordinates.
+
+    Parameters
+    ----------
+    xccd : np.ndarray
+        Array of x CCD pixel coordinates.
+    yccd : np.ndarray
+        Array of y CCD pixel coordinates.
+    wcs : lsst.afw.image.Wcs
+        WCS object for the guider detector.
+    obs_time : astropy.time.Time
+        Observation time.
+
+    Returns
+    -------
+    alt, az : np.ndarray
+        Arrays of alt and az coordinates (degrees).
+    """
+    if len(xccd) > 0:
+        az, alt = convert_pixels_to_altaz(wcs, obs_time, xccd, yccd)
+    else:
+        alt, az = np.array([]), np.array([])
+
+    return alt, az
+
+
+def convert_roi_to_ccd(
+    xroi: np.ndarray, yroi: np.ndarray, guiderData: GuiderData, guiderName: str
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Convert roi coordinates to CCD pixel coordinates.
+
+    Parameters
+    ----------
+    xroi : np.ndarray
+        Array of x roi coordinates (pixels within the ROI).
+    yroi : np.ndarray
+        Array of y roi coordinates (pixels within the ROI).
+    guiderData : GuiderData
+        GuiderData object containing the guider datasets and view information.
+    guiderName : str
+        Name of the guider (e.g., 'R44_SG0', etc.)
+
+    Returns
+    -------
+    xccd, yccd : np.ndarray
+        Arrays of CCD pixel coordinates.
+    """
+    view = guiderData.view
+    stamps = guiderData.datasets[guiderName]
+
+    if np.isscalar(xroi):
+        xroi = np.array([xroi])
```

Comment:
I'm not certain, but I think you can replace the 2-line check + creation with just `xroi = np.asarray(xroi)`

Reply:
got it.